man/systemd-cryptsetup-generator.xml

```diff
@@ -165,6 +165,80 @@
         the initrd.</para>
         </listitem>
       </varlistentry>
+
+      <varlistentry>
+        <term><varname>luks.header_dev=</varname></term>
+        <term><varname>rd.luks.header_dev=</varname></term>
+
+        <listitem><para>Takes a LUKS super block UUID followed by a <literal>=</literal>
+        and a block device specification with filesystem containing LUKS header.</para>
```

Comment:
this is confusing. The specified block device does *not* contain a file system, but this sentence suggests so.

Reply:
Maybe: "Configures a device where to find the LUKS header belonging to an encrypted file system, in case the LUKS header detached, i.e. is not placed on the encrypted file system block device itself. Takes a LUKS superblock UUID followed by <literal>=</literal> and a block device specification indicating the location of the LUKS header."

or so

Reply:
> this is confusing. The specified block device does _not_ contain a file system, but this sentence suggests so.

Because it should suggest it:). The purpose of luks.header_dev= is exactly to specify external block device containing a filesystem with detached LUKS header. If it was just plain bock device with detached LUKS header placed in the beginning it could have been placed directly in existing luks.options like this: "luks.options=header=/dev/sda42"

Reply:
Hmm, so I think the confusion stems from the fact that the location of the header is specified in two separate steps: `luks.header-dev=` and `header=...`. This will be annoying to use, also because the luks UUID will need to be specified in both keys. Why not extends the header= parameter to take the device specification before the path? We use this in other similar circumstances (@msekletar added some code like this semi-recently), so it'd be a matter of reusing this code. This will be much more logical because there will be a single parameter specifying where to find the header.